import "DataCommon.proto";

package ProtoMine;

message MineItem
{
	required uint32	uid 		= 1;  //矿主uid =0则表示无人占领
	required uint32 level 		= 2;  //矿等级
	required uint32	donate_cnt	= 3;  //已捐献次数
	required bool	is_donate	= 4;  //自己是否捐献
	required bytes	name		= 5;  //矿主名字 
}

/*
//enum MineOp
//{
//	e_OpenHole = 1, //开矿
//	e_Donate   = 2,   //援建
//	e_MinersHarvest = 3, //矿主收获
//	e_MineReward	= 4, //国家矿的奖励
//};
 */

message MineRecord
{
	required bytes	operate		= 1;  //操作者
	required bytes	target		= 2;  //目标者
	required uint32	op			= 3;  //操作
	required uint32	type		= 4;  //0--普通矿洞  1--国家矿洞
	repeated uint32	params		= 5;  //参数列表
	required uint32	ts			= 6;  //时间戳
}

/////////////////////////////////////////////////////////////
//矿洞信息
message MineInfoReq
{
}

message MineInfoResp
{
	required	uint32	 mine_exp		= 1;
	required    uint32	 mine_level		= 2;
	required	MineItem kingdom_mine	= 3;  //国家矿洞
	repeated	MineItem normal_mine	= 4;  //普通矿洞
	required    uint32 	 mine_normal_ts  = 5;  //普通矿开矿时间
	required 	uint32	 mine_reward_ts	 = 6;  //国家矿领取时间
}
//////////////////////////////////////////////////////////////
//开普通矿
message MineNormalOpenReq
{
	required 	uint32	index = 1; //1-8 偏移位置
}

//国家矿
message MineKingdomOpenReq
{
}

message MineOpenResp
{
	required  	MineItem 	 				mine	= 1;
	required 	DataCommon.UserResourceCPP	assets	= 2;
	required	uint32	 					mine_exp 	= 3;
	required    uint32	 					mine_level	= 4;
	optional    uint32 	 					mine_normal_ts  = 5;  //普通矿开矿时间
}
//////////////////////////////////////////////////////////////
//捐赠
message DonateNormalReq
{
	required 	uint32	index = 1; //1-8 偏移位置
}

message DonateKingdomReq
{
}

message DonateResp
{
	required  	MineItem 	 				mine	= 1;
	required 	DataCommon.UserResourceCPP	assets	= 2;
	required	uint32	 					mine_exp 	= 3;
	required    uint32	 					mine_level	= 4;
}
//////////////////////////////////////////////////////////////
//领取国矿奖励
message MineKingdomRewardReq
{
	
}

message MineKingdomRewardResp
{
	required uint32    					op_ts = 1;  //上次领奖时间 
	required DataCommon.UserResourceCPP	assets	= 2; //
}

//国家矿矿主领奖
message MineKingdomMinerRewardReq
{
}

message MineKingdomMinerRewardResp
{
	required uint32    				  op_ts = 1;  //上次领奖时间 
	required DataCommon.RewardItemCPP exp = 2; //经验
}

////////////////////////////////////////////////////////////
message MineRecordReq
{
}

message MineRecordResp
{
	repeated MineRecord records = 1;
}

//
message SyncUserBaseInfo
{
	required DataCommon.CommonBaseRewardCPP info = 1;
	required uint32							level = 2;
}
